### DISCUSSION TITLE
Views not found after moving them from the root project to a module project

### DISCUSSION THREAD

**Discussion by maglini** [OrchardCMS/OrchardCore#9909] [Q&A] [answered]
We created module with next startup 
![image](https://user-images.githubusercontent.com/60664635/125461142-b7ea8752-3985-4932-8821-c28eefddaecf.png)
The next structure project: 
![image](https://user-images.githubusercontent.com/60664635/125461462-7d8708e4-7ab5-4ae9-bb69-7a49c4363548.png)
And we cant see our view 
![image](https://user-images.githubusercontent.com/60664635/125461577-d2bf1976-d0d4-47ca-8595-0b3a21ce9d24.png)

**Comment by ns8482e** [comment]
Did you add Module Target to your module project? and your module project as project reference to your web application?

**Reply by xperiandri** [reply]
Yes, we did. The issue is that after calling `MapAreaControllerRoute` within a module views are being looked for in `Area\Views` folder instead of project root folder `\Views`.
What can be wrong that it happens this way?

**Reply by xperiandri** [reply]
@maglini post csproj, please

**Reply by xperiandri** [reply]
And Auth\Startup.cs

**Reply by ns8482e** [reply]
OC will place all  Modules and Application under `Area`. So here for module `Bitrix24.Tenant` with  `SettingController` for `Index` action,  the default url is `/Bitrix24.Tenant/Setting/Index`. By setting `MapAreaControllerRoute` you can change the pattern.

See how it's setup in Demo Module

https://github.com/OrchardCMS/OrchardCore/blob/a779df8beb42977efbb053a12e6fd7daac92810e/src/OrchardCore.Modules/OrchardCore.Demo/Startup.cs#L43-L48

**Reply by xperiandri** [reply]
As you can see from the screenshot above we did exactly the same and we navigate to the URL `/Bitrix24.Tenant/settings` which is covered by paattern

**Reply by ns8482e** [reply]
Do you get this error on Dev or in Prod? Also Are you using source code or nuget (what version) ?

**Reply by xperiandri** [reply]
NuGet

**Reply by xperiandri** [reply]
Dev

**Reply by maglini** [reply]
Our module Bitrix24.Tenant csproj 
![image](https://user-images.githubusercontent.com/60664635/125693535-a45634fe-5954-4f1d-ac35-619a04960776.png)

**Reply by maglini** [reply]
Auth Startup 
![image](https://user-images.githubusercontent.com/60664635/125693796-1f4df366-5c5f-4255-9cf1-abac6690eec6.png)

**Reply by ns8482e** [reply]
I guess you don’t need `AddRazorPages` as `AddMvc` does it for you

**Reply by ns8482e** [reply]
Remove line 119 to 131, Don't use `UseRouting` and `UseEndpoints` before `UseOrchardCore`, this resulted in resolving the Controller/Action in non-tenant aware fashion. ( that's why it's looking views under "Areas" folder physically )

Instead use following to configure in Tenant pipeline
```
services
  .AddOrchardCore()
  .ConfigureServices( s=> )
  .Configure( (app,routes,srv) => routes.MapAreaController() );
```

**Reply by xperiandri** [reply]
``` C#
services.AddOrchardCore()
    .AddMvc()
    .WithTenants()
    .ConfigureServices(s =>
    {
        s.AddRazorPages();
        s.AddServerSideBlazor();
    })
    .Configure((app, routes, srv) =>
    {
        app.UseRouting();

        app.UseEndpoints(endpoints =>
        {
            endpoints.MapControllerRoute(
                name: "default",
                pattern: "{controller=Home}/{action=Index}/{id?}");

            endpoints.MapBlazorHub();

            //endpoints.MapFallbackToPage("/_Host");
        });

    });
```

@ns8482e Blazor works, controller does not. What is wrong?

**Reply by jtkech** [reply]
Anyway don't call `app.UseRouting()` and `app.UseEndpoints()`.
It is automatically done when building the tenant pipeline.

So try by just using `routes.MapControllerRoute()`

Also the 2 `OrchardCoreBuilder` helpers `ConfigureServices()` and `Configure()` allow to pass an order parameter, the execution order may be important in your case, this parameter has the same effect as the `Order` and `ConfigureOrder` properties of a module startup.

Note: Normally you don't need to call `AddRazorPages()`.

**Reply by xperiandri** [reply]
I tried to use `routes` instead but with no luck. I will check how can I use order

**Reply by ns8482e** [reply]
Looks like your module route conflict with Home Route feature, Disable Home Route feature.

**Reply by xperiandri** [reply]
Is it awailable in application targets?
We do not reference CMS targets

**Reply by ns8482e** [reply]
Home Route is CMS module.

**Reply by xperiandri** [reply]
So for now it works only if we add route within a module with `MapAreaControllerRoute`, not root app

**Comment by xperiandri** [comment, marked as answer]
The issue was caused by incorrect build actions for razor files after copy/move from the root project. So we have to remove that build action from the module project for razor files.
